File: p2/p2.pl
#!/usr/bin/perl -w

use strict;


sub main{
    my @fib_array = (1,2);
    my $sum = 0;
    while(1){
        my $new_value = $fib_array[$#fib_array] + $fib_array[$#fib_array-1];
        last if $new_value > 4000000;
        push @fib_array, $new_value;
    }
    for(grep{ !($_%2) } @fib_array){ $sum += $_ }
    print "Answer:  $sum \n"
}

main();

File: p2/p2.py
#!/usr/bin/python

def main():
    answer = 2
    fib_a  = 2
    fib_b  = 1

    while fib_a < 4000000:
        fib_a, fib_b = fib_a + fib_b, fib_a
        if fib_a % 2 == 0: answer += fib_a 

    print "Answer %d " % answer

main()
